### ISSUE TITLE
Calendar click for multiple assignments on same day

### ISSUE THREAD

**Issue by lucastbelem** [jesus-collective/mobile#795]
Hey @JonEsparaz. What did you do for this case? When a date has 2 lessons/assignments associated with them, what did you tell the calendar to take the user to on click? First to occur based on time?

Here is an idea with less than 2 minutes of thinking so it might be a bad one...

What if when there are 2 assignments on one day, when the user clicks that date on the calendar, it gives them a tiny popup on the calendar with a list of the 2 or 3 assignments for them to click to?

**Comment by JonEsparaz** [comment]
There wasn't much clarity on this behaviour, so there's actually no logic.

It will get the lesson that was last inputted into the database (excluding YouTube items) for each day.

**Comment by JonEsparaz** [comment]
The [agenda component](https://github.com/wix/react-native-calendars#agenda) from `react-native-calendars` might be a good alternative to what you are describing (checkout the GIF at the above link to see an example).

**Comment by lucastbelem** [comment]
@JonEsparaz this works for me. What are the chances you can work on this for us, Jon?

**Comment by JonEsparaz** [comment]
@lucastbelem I can more than likely implement this.

Questions:
- @GeorgeBellTMH is there a particular reason why the current to-do list and calendar are hidden?
- Can we just have the agenda component (no separate calendar and to-do list) since it is essentially a combination of the calendar and to-do list?

**Comment by lucastbelem** [comment]
@JonEsparaz they're hidden right now for 2 reasons (but mainly because courses are live and we don't users to deal with it broken): 
To dos stopped working (blank screen again). If you're available tonight and want/can tackle this, that'd be great. George can then focus on some messaging issues we're having.

Calendars - it's my fault. I should've got things moving with you on this as soon as you posted a solution, but I dropped the ball on this one. Now they're saying "it's not working, it's not taking users to the right lesson". So I thought we should hide it until we "fix" it.

As for combining the 2, I like the idea, but the stakeholders for courses are pretty sensitive, so I'd need to check with them first and make sure they understand it.

**Comment by lucastbelem** [comment]
I'm gonna ask this tonight and see how quickly they get back to me.

**Comment by JonEsparaz** [comment]
> To dos stopped working (blank screen again). If you're available tonight and want/can tackle this, that'd be great.

I don't necessarily have time tonight, but I can provide some insight. Bad array arithmetic is certainly the cause of this problem. There's probably off-by-one errors hiding in the code or the week/lesson numbers are getting out of sync with the array indices. Whatever the case, refactoring/reimplementing with objects and accessing/updating the data based on ID instead of index will be foolproof.

**Comment by GeorgeBellTMH** [comment]
Yup, this is definitely the situation. The whole thing needs a bunch of rework to resolve the sorting/filtering issues and indexing issues. Definitely the way to fix it is with ids instead of indexes.

**Comment by JonEsparaz** [comment]
FYI: The Agenda component turns out to have very poor support for React Native Web (ugh).

Thus, I've implemented what @lucastbelem originally suggested. If there's only one lesson for the date marked on the calendar, clicking on it takes you directly to the lesson. Otherwise, you see this list of clickable lessons, which renders on top of the calendar: 
![calednar popop](https://user-images.githubusercontent.com/48423418/113968048-4f224500-9800-11eb-832d-5e149b858e00.JPG)

**Comment by JonEsparaz** [comment]
Re-opening as I suspect the UI/UX isn't finalized.

@lucastbelem another idea: when there are multiple lessons on one day, clicking on the calendar takes you to the course detail screen, but filters the week's lessons by date, so you only see the lessons for that day. Thoughts?

**Comment by JonEsparaz** [comment]
@lucastbelem feedback here would be good since I'll tackle a few other items tonight.

**Comment by lucastbelem** [comment]
@JonEsparaz dude! Sorry I missed this. 

If I follow your suggestion here, there would be no need for the little popup thing. True?
Then if it's 2 lessons in a day, it goes to a page where the lessons for that day live, but assignments from a different day don't show up. Is that right?

If so, I like it!

Also, can you take a look at this:

Can we:

 Make the legend work so that the coloured dots match the type of assignment on a date?
 For days with more than 1 lesson type, can we make a dot that is filled half with one colour and half with the other colour? Is this too overkill? Too little? @mmacedom do you have thoughts on this from a UI standpoint?
<img width="321" alt="Screen Shot 2021-04-12 at 6 22 40 PM" src="https://user-images.githubusercontent.com/48295718/114471242-434ad000-9bbe-11eb-8119-b2f5e6edf390.png">

**Comment by lucastbelem** [comment]
@JonEsparaz seems we're moving too fast haha

For the legend, you suggested multiple dots. That works fine for me :)

**Comment by JonEsparaz** [comment]
> If I follow your suggestion here, there would be no need for the little popup thing. True?
> Then if it's 2 lessons in a day, it goes to a page where the lessons for that day live, but assignments from a different day don't show up. Is that right?

@lucastbelem yes, looks like we're on the same page about this and the dots on the calendar - will implement today or tomorrow.

**Comment by lucastbelem** [comment]
Fantastic!

**Comment by mmacedom** [comment]
@lucastbelem @JonEsparaz If you guys want my two cents on the issue, I think the solution proposed (multiple dots) is the best way to move forward in this situation if we are not looking to heavily modify the component. Despite the fact, I think some users may have a hard time seeing the dots because they are very small. Additionally, there's the issue of color blind users not being able to distinguish between the colors of the dots especially at that size. Unfortunately, I don't have a better solution unless we modify the calendar completely.

**Comment by lucastbelem** [comment]
@mmacedom Thanks! We'll go with the dots for now and take a look later when we work on a "facelift" for the platform :)
